### PR TITLE
[Bug Fix] Fix typo in SpatialSeparableConvoluiton layer name and add related docs

### PR DESCRIPTION
**Pull request by yiheng**
## What changes were proposed in this pull request?
Fix typo in SpatialSeparableConvoluiton layer name and add related docs

## How was this patch tested?
unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2131

### CODE REVIEW

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 1]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
   [-0.38188276 -0.36746511 -0.37627563 -0.34141305]]]
 ```
 
+## SpatialSeperableConvolution ##
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: Typo.... Separable

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 2]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+
+**Scala:**
+```scala
+val m  = SpatialSeperableConvolution(nInputChannel, nOutputChannel, depthMultiplier, kW, kH, sW, sH, pW, pH, hasBias, dataFormat, wRegularizer = null,bRegularizer = null, pRegularizer = null, initDepthWeight = null, initPointWeight = null, initBias = null)
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: Separ...

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 3]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+```
+**Python:**
+```python
+m = SpatialFullConvolution(n_input_channel, n_output_channel, depth_multiplier, kernel_w, kernel_h, stride_w = 1, stride_h = 1, pad_w = 0, pad_h = 0, with_bias = True, data_format="NCHW", w_regularizer=None, b_regularizer=None, p_regularizer=None)
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: FullConvolution??

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 2 above)
Comment (hkvision) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: Any default value for pW, pH, hasBias, dataFormat?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 4]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+ * `nInputChannel` The number of expected input planes in the image given into forward()
+ * `nOutputChannel` The number of output planes the convolution layer will produce.
+ * `depthMultiplier` how many internal channels are generated per input channel
+ * `kW` The kernel width of the convolution.
+ * `kH` The kernel height of the convolution.
+ * `sW` The step of the convolution in the width dimension.
+ * `sH` The step of the convolution in the height dimension.
+ * `pW` The additional zeros added per width to the input planes. Default is 0.
+ * `pH` The additional zeros added per height to the input planes. Default is 0.
+ * `hasBias` do we use a bias on the output
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: default?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 5]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+ * `dataFormat` image data format, which can be NHWC or NCHW
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: default value?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 6]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+scala> print(gradOut)
+(1,1,.,.) =
+0.1     0.1
+0.1     0.1
+
+(1,2,.,.) =
+0.1     0.1
+0.1     0.1
+
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcF$sp of size 1x2x2x2]
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: what is $mcF$sp?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 7]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+tensor_input = np.random.rand(1, 1, 3, 3)
+print "input is :",tensor_input
+out = m.forward(tensor_input)
+print "output m is :",out
+```
+Gives the output,
+
+```python
+creating: createSpatialFullConvolution
+--------- tensor input---------
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: What is the red line?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md [hunk 8]
```diff
@@ -802,6 +802,124 @@ output m is : [[[-0.04712385  0.21949144  0.0843184   0.14336972]
...
+
+
+**Python example:**
+```python
+from bigdl.nn.layer import *
+import numpy as np
+
+m = SpatialSeperableConvolution(1, 2, 1, 2, 2)
+
+print "--------- tensor input---------"
```
Comment (hkvision) [inline docs/docs/APIGuide/Layers/Convolution-Layers.md]: remove this?

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 2 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: let me make it consist with python

Comment (hkvision) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: but such code written in Python is not well formatted....

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 3 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: done

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 8 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: done

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 7 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: removed

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 6 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: it means some specific version of class in scala. which is a part of class name when print the tensor

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 4 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: done

File: docs/docs/APIGuide/Layers/Convolution-Layers.md (hunk 1 above)
Comment (yiheng) [reply docs/docs/APIGuide/Layers/Convolution-Layers.md]: thanks for pointing it out. changed

**Comment by yiheng** [comment]
jenkins passed

**Pull request merged**